### PR TITLE
Fix yarn warnings

### PR DESCRIPTION
**Pull request by gnidan**
Firstly, this fixes the circular dependency warnings by making the following packages into peer dependencies:
- truffle-resolver's truffle-contract
- truffle-decoder's truffle

Additionally, this fixes warnings about inconsistent node executables by finding and replacing all instances of package.json files defining scripts in terms of `npm run <script>`. `yarn <script>` is used instead in all cases now.

### CODE REVIEW

File: packages/truffle/scripts/test.sh [hunk 1]
```diff
@@ -3,9 +3,9 @@
 set -o errexit
 
 if [ "$GETH" == true ]; then
-  npm run build-cli && mocha --timeout 50000 --grep @ganache --invert --colors $@
+  yarn build-cli && mocha --timeout 50000 --grep @ganache --invert --colors $@
 elif [ "$COVERAGE" == true ]; then
   NO_BUILD=true mocha --no-warnings --timeout 7000 --grep @geth --invert --colors $@
 else
-  npm run build-cli && mocha --no-warnings --timeout 7000 --grep @geth --invert --colors $@
+  yarn run build-cli && mocha --no-warnings --timeout 7000 --grep @geth --invert --colors $@
```
Comment (eggplantzzz) [inline packages/truffle/scripts/test.sh]: Does this need to have 'run' removed?

**Comment by eggplantzzz** [comment]
Other than the one comment, it looks cool.

File: .travis.yml [hunk 2]
```diff
@@ -26,17 +26,16 @@ matrix:
...
 
-
 install:
-  - npm run bootstrap
+  - yarn bootstrap
 
 cache:
   directories:
     - node_modules
     - $(npm config get prefix)/lib/node_modules # globally installed stuff (i.e. lerna)
```
Comment (CruzMolina) [inline .travis.yml]:
```suggestion
    - $(yarn config get prefix)/lib/node_modules # globally installed stuff (i.e. lerna)
```

File: README.md [hunk 3]
```diff
@@ -57,7 +57,7 @@ We welcome pull requests. To get started, just fork this repo, clone it locally,
 # Install
 npm install -g lerna@3.4.3
 npm install -g yarn
-npm run bootstrap
+yarn bootstrap
 
 # Test
 npm test
```
Comment (CruzMolina) [inline README.md]:
```suggestion
yarn test
```

File: packages/truffle-contract/package.json [hunk 4]
```diff
@@ -4,7 +4,7 @@
   "description": "A better contract abstraction for Ethereum (formerly EtherPudding)",
   "main": "index.js",
   "scripts": {
-    "prepare": "npm run compile",
+    "prepare": "yarn compile",
     "publish:next": "node ../truffle/scripts/prereleaseVersion.js next next",
     "test": "./scripts/test.sh",
     "test:debug": "$(npm bin)/mocha --inspect-brk",
```
Comment (CruzMolina) [inline packages/truffle-contract/package.json]:
```suggestion
     "test:debug": "$(yarn bin)/mocha --inspect-brk",
```
Same for `"test:trace"` below.

File: packages/truffle-decoder/package.json [hunk 5]
```diff
@@ -50,5 +49,8 @@
     "truffle-decode-utils": "^1.0.11",
     "web3": "1.0.0-beta.37"
   },
+  "peerDependencies": {
```
Comment (CruzMolina) [inline packages/truffle-decoder/package.json]: 🎉

File: packages/truffle/scripts/test.sh (hunk 1 above)
Comment (CruzMolina) [inline packages/truffle/scripts/test.sh]:
```suggestion
  yarn build-cli && mocha --no-warnings --timeout 7000 --grep @geth --invert --colors $@
```

Comment (gnidan) [reply packages/truffle/scripts/test.sh]: Yep, good catch!

File: .travis.yml (hunk 2 above)
Comment (gnidan) [reply .travis.yml]: Hm I just tried this locally and it doesn't seem to work. Have you tried it?

Comment (CruzMolina) [reply .travis.yml]: Hmm, seems to work on my end?

```
➜  truffle git:(quorum-subprovider) ✗ $(yarn config get prefix)/lib/node_modules
➜  node_modules ls
yarn
➜  node_modules 
```

Comment (gnidan) [reply .travis.yml]: What happens if you run `yarn config get prefix` by itself? I get `undefined`

Comment (CruzMolina) [reply .travis.yml]: same, which is why we pass the expected path where the global `node_modules` would be installed/cached.

**Comment by haltman-at** [comment]
Question: When did `truffle-decoder` gain a dependency on `truffle`?  I remember seeing such warnings starting to pop up, but I don't remember ever deliberately adding that.  Does it actually need to be there?  Did someone else add it?

**Comment by gnidan** [comment]
@haltman-at it's a dev. dependency, really, since the decoder tests run Truffle itself, I believe.

**Comment by haltman-at** [comment]
@gnidan Oh, duh.  That makes sense.  Guess that was already there then and it was something else that changed to form the loop.

File: .travis.yml (hunk 2 above)
Comment (gnidan) [reply .travis.yml]: This seems wonky with how it returns undefined. We'll just have to keep an eye to make sure the Travis cache behaves.

**Comment by gnidan** [comment]
Thanks all!

**Comment by CplXhristianUSMC** [comment]
Thank you so much for your reply I'm sorry that it's taking me so long to
get back to your emails currently I'm in the process of moving and will be
driving no less than 1300 miles tonight across state lines to get to my new
house please forgive any late replies I greatly appreciate any and all work
that you have been doing to help rectify this matter truly thankful and
gratefully yours cplxhristianusmc

**Pull request merged**